app/api/auth.py

```diff
@@ -305,7 +305,7 @@ async def verify_token(self, token: str) -> dict[str, Any]:
                     rsa_key,
                     algorithms=["RS256"],
                     audience=self.api_audience,
-                    issuer=f"https://sts.windows.net/{self.tenant_id}/",
+                    issuer=f"https://{self.tenant_id}.ciamlogin.com/{self.tenant_id}/v2.0",
```

Comment:
This is very poorly documented, but External tenants need this format for auth to work: https://learn.microsoft.com/en-us/entra/identity-platform/security-tokens#token-endpoints-and-issuers